sample-notificator-domain/src/main/groovy/life/qbic/business/notification/refactor/SendEmail.java

```diff
@@ -0,0 +1,51 @@
+package life.qbic.business.notification.refactor;
+
+import java.util.List;
+import java.util.function.Consumer;
+import life.qbic.business.notification.create.NotificationContent;
+
+/**
+ * <p>Implements the business logic of sending emails for project updates.</p>
+ */
+public class SendEmail implements Consumer<List<NotificationContent>>, SendEmailInput{
+    private final EmailSender<NotificationEmail> sendOrRemember;
+    private final FailureEmailSender informAdmin;
+    private final EmailGenerator<NotificationEmail> generateEmail;
```

Comment:
Since we have a lot of different names for the same thing I would suggest to not introducing new names here.
The interface is for the use case, right? I find it confusing if the name of the interface is different and implies functionality that is not implied by the interface name.

Reply:
I would like to keep the class names as is as I think it is reasonably understandable. In case you were referring to the variable names; I chose them so that the code reads like a sentence and they contain even more specific meaning than their class names. The `informAdmin` for example only hints towards what the role of the failure email sender is in this process/pipeline. It still sends failure emails. The same is true for the `sendOrRemember` as it highlights that this sends emails or remembers them for later. The `EmailSender` still sends the email regardless of whether it's remembered emails are of interest. The `generateEmail` is pretty self-explanatory and an `EmailGenerator` should generate emails IMHO.

Reply:
I see why you did it but from my point of view I dont think that it is such a good idea because it seems to me that by naming it like this you make assumptions about what the implementing class does. And if the concret implementation actually is what the class does I dont see why its not reasonable to name the class like this. But anyhow, I will not block this PR because of that.